File: Editorial/0000-0100/0069_Sqrt_x/CSharp/3_Recursion_Bit_Shifts.cs
public class Solution {
    public int MySqrt(int x) {
        if (x < 2)
            return x;
        int left = MySqrt(x >> 2) << 1;
        int right = left + 1;
        return (long)right * right > x ? left : right;
    }
}

File: Editorial/0000-0100/0069_Sqrt_x/Go/3_Recursion_Bit_Shifts.go
// Golang translation
func mySqrt(x int) int {
    if x < 2 {
        return x
    }
    left := mySqrt(x>>2) << 1
    right := left + 1
    if right*right > x {
        return left
    }
    return right
}
